### PR TITLE
IT-3860: Pass secure parameter to automation script

### PR DESCRIPTION
**Pull request by brucehoff**
While CloudFormation [allows invoking SSM Parameter look up when passing parameters](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references-ssm-secure-strings.html) , Organization Formation fails with the error:

```
SSM Secure reference is not supported in: [AWS::Lambda::Function/Properties/Environment/Variables/ScriptParameters]

```

The fix is instead to simply pass the SSM Parameter name to the Automation script and update the Lambda to retrieve the secure parameter(s) to pass to the Automation Document.

This PR also updates the documentation to explain how to use the script.

### CODE REVIEW

File: org-formation/090-systems-manager/Scheduled-Script-Automation.yaml [hunk 1]
```diff
@@ -50,8 +50,8 @@ Parameters:
   ScriptParameters:
     Type: String
     Description: >-
-      space delimited list of environment name-value pairs in the format name:value,
-      e.g., ENVNAME1:envval1 ENVNAME2:envval2
+      space delimited list of environment name-ssm parameter store pairs in the format env-name:ssm-name,
```
Comment (zaro0508) [inline org-formation/090-systems-manager/Scheduled-Script-Automation.yaml:53]: hmm, every parameter needs to have a corresponding entry in the SSM?  What if you want to define parameters that are not secrets?

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: org-formation/090-systems-manager/Scheduled-Script-Automation.yaml [hunk 2]
```diff
@@ -180,6 +185,20 @@ Resources:
               AutomationDocumentScriptExecution=os.environ['AutomationDocumentScriptExecution']
               ScriptUrl=os.environ['ScriptUrl']
               ScriptParameters=os.environ['ScriptParameters']
+              ScriptParametersArray = str(ScriptParameters).split()
+              ScriptKeyValueParameters = None
+              for envParamNameString in ScriptParametersArray:
+                envParamNamePair = envParamNameString.split(":")
```
[the remark this answers is not in the document]
Comment (brucehoff) [reply org-formation/090-systems-manager/Scheduled-Script-Automation.yaml:191]: The "ssm portion" should not contain an ":" following our naming convention.  If it does then line 193 will raise an exception.

File: org-formation/090-systems-manager/Scheduled-Script-Automation.yaml [hunk 3]
```diff
@@ -180,6 +185,20 @@ Resources:
...
+                if len(envParamNamePair) != 2:
+                    raise Exception(f"Expected key:value pair but got {envParamNameString}")
+                envVarName=envParamNamePair[0]
+                ssmParamName=envParamNamePair[1]
+                envVarValue = client.get_parameter(Name=ssmParamName, WithDecryption=True)['Parameter']['Value']
+                keyValuePair=envVarName+":"+envVarValue
```
[the remark this answers is not in the document]
Comment (brucehoff) [reply org-formation/090-systems-manager/Scheduled-Script-Automation.yaml:197]: If so then the script will get a blank value for the given env var.  That will only happen if someone enters an empty value in SSM Parameter Store.

File: org-formation/090-systems-manager/Scheduled-Script-Automation.yaml (hunk 1 above)
Comment (brucehoff) [reply org-formation/090-systems-manager/Scheduled-Script-Automation.yaml:53]: Today the non-secret param's would be stored as secrets.  If that becomes limiting we can modify this automation.

File: org-formation/090-systems-manager/Scheduled-Script-Automation.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (zaro0508) [reply org-formation/090-systems-manager/Scheduled-Script-Automation.yaml:191]: the `ssm portion` i'm referring to is the value that the user saved to in the SSM.  For example, lets say someone saved `PasswordContains:Colon` to  `/MySQL/password` in the SSM.

File: org-formation/090-systems-manager/Scheduled-Script-Automation.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (zaro0508) [reply org-formation/090-systems-manager/Scheduled-Script-Automation.yaml:197]: what if they forget to define the parameter in the AWS SSM?  or they define it with an incorrect parameter name?

File: org-formation/090-systems-manager/Scheduled-Script-Automation.yaml (hunk 1 above)
Comment (zaro0508) [reply org-formation/090-systems-manager/Scheduled-Script-Automation.yaml:53]: Not a fan, but i guess we can leave until later.

File: org-formation/090-systems-manager/Scheduled-Script-Automation.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (zaro0508) [reply org-formation/090-systems-manager/Scheduled-Script-Automation.yaml:191]: i don't see any docs for naming convention but i guess it's probably unlikely ssm params contains colons.

File: org-formation/090-systems-manager/Scheduled-Script-Automation.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (brucehoff) [reply org-formation/090-systems-manager/Scheduled-Script-Automation.yaml:197]: In that case line 196 will raise an exception, i.e., `get_parameter` will fail.

**Review by zaro0508 — APPROVED** [review]

**Pull request merged**